src/org/omegat/gui/exttrans/MachineTranslationInfo.java

```diff
@@ -32,12 +32,26 @@
  * @author Alex Buloichik (alex73mail@gmail.com)
  * @author Aaron Madlon-Kay
  */
-public class MachineTranslationInfo {
+public class MachineTranslationInfo implements Comparable<MachineTranslationInfo> {
     public final String translatorName;
     public final String result;
 
     public MachineTranslationInfo(String translatorName, String result) {
         this.translatorName = translatorName;
         this.result = result;
     }
+
+    /**
+     * Compares this object with the translator name, then result.
+     * Returns a negative integer, zero, or a positive integer
+     * as this object is less than, equal to, or greater than the specified object.
+     */
+    @Override
+    public int compareTo(final MachineTranslationInfo o) {
+        int compare = this.translatorName.compareTo(o.translatorName);
+        if (compare == 0) {
+            return this.result.compareTo(o.result);
+        }
+        return compare;
```

Comment:
I feel like usually this is done with the comparison flipped:

```suggestion
        int compare = translatorName.compareTo(o.translatorName);
        if (compare != 0) {
            return compare;
        }
        return result.compareTo(o.result);
```

(Also you don't need `this` for the current object's members since there is no shadowing.)

Reply:
I've accepted the proposal.
These `hashCode` and `compareTo` were generated by IDE, intelliJ.